service/openapi/src/main/java/org/dotwebstack/framework/service/openapi/query/GraphQlQueryBuilder.java

```diff
@@ -29,38 +40,41 @@
       return Optional.empty();
     }
 
-    ResponseTemplate okResponse = responseSchemaContext.getResponses()
-        .stream()
-        .filter(r -> r.getResponseCode() == 200)
-        .findFirst()
-        .orElseThrow(() -> new InvalidConfigurationException("No OK response found"));
+    ResponseTemplate okResponse = getOkResponse(responseSchemaContext);
+
+    Optional<Field> rootField = OasToGraphQlHelper.toQueryField(queryName, okResponse, inputParams, this.pagingEnabled);
+    if (rootField.isEmpty()) {
+      return Optional.empty();
+    }
+    GraphQlQuery query = toQueryInput(rootField.get());
 
-    List<Field> fields = OasToGraphQlHelper.toQueryFields(okResponse, inputParams);
-    Optional<GraphQlQuery> query = toQueryInput(queryName, fields);
-    query.ifPresent(q -> addKeys(q, dwsQuerySettings.getKeys(), inputParams));
+    addKeys(query, dwsQuerySettings.getKeys(), inputParams);
     Map<String, Object> variables;
-    variables = query.map(graphQlQuery -> addFilters(graphQlQuery, responseSchemaContext.getDwsQuerySettings()
-        .getFilters(), inputParams))
-        .orElseGet(Map::of);
-    query.ifPresent(q -> addFilters(q, dwsQuerySettings.getFilters(), inputParams));
+    variables = addFilters(query, responseSchemaContext.getDwsQuerySettings()
+        .getFilters(), inputParams);
+    addFilters(query, dwsQuerySettings.getFilters(), inputParams);
+    addPaging(query, dwsQuerySettings.getPaging(), inputParams);
 
-    return query.map(q -> QueryInput.builder()
-        .query(q.toString())
+    return Optional.of(QueryInput.builder()
+        .query(query.toString())
         .variables(variables)
         .build());
-
   }
 
-  private Optional<GraphQlQuery> toQueryInput(String queryName, List<Field> fields) {
-    Field root = new Field();
-    root.setChildren(fields);
-    root.setName(queryName);
-
+  private GraphQlQuery toQueryInput(Field rootField) {
     GraphQlQuery.GraphQlQueryBuilder builder = GraphQlQuery.builder();
-    builder.field(root);
+    builder.field(rootField);
     builder.queryName("Query");
-    return Optional.of(builder.build());
+    return builder.build();
+
+  }
 
+  private ResponseTemplate getOkResponse(ResponseSchemaContext responseSchemaContext) {
+    return responseSchemaContext.getResponses()
+        .stream()
+        .filter(r -> r.getResponseCode() == 200)
+        .findFirst()
+        .orElseThrow(() -> new InvalidConfigurationException("No OK response found"));
```

Comment:
Static import van ExceptionHelper:

```suggestion
        .orElseThrow(() -> invalidConfigurationException("No OK response found"));
```